### ISSUE TITLE
Building error on Cori

### ISSUE THREAD

**Issue by jychoi-hpc** [ornladios/ADIOS2#1459]
I am trying to build Adios2 with GNU on Cori but getting the following error:
```
/usr/bin/ld: ../../lib64/libtaustubs.a(tautimer.cpp.o): in function `open_preload_libraries()':
/global/cscratch1/sd/jyc/src/adios2-devel/source/adios2/toolkit/profiling/taustubs/tautimer.cpp:112: undefined reference to `dlopen'
/usr/bin/ld: ../../lib64/libtaustubs.a(tautimer.cpp.o): in function `assign_function_pointers()':
/global/cscratch1/sd/jyc/src/adios2-devel/source/adios2/toolkit/profiling/taustubs/tautimer.cpp:121: undefined reference to `dlsym'
/usr/bin/ld: /global/cscratch1/sd/jyc/src/adios2-devel/source/adios2/toolkit/profiling/taustubs/tautimer.cpp:136: undefined reference to `dlsym'
```

Here is my cmake command:
```
cmake -DCMAKE_INSTALL_PREFIX=/global/cscratch1/sd/jyc/sw/adios2/v2.1.0-2939-g421f29d7/knl/gnu -DCMAKE_BUILD_TYPE=RelWithDebInfo -DBUILD_SHARED_LIBS=OFF -DADIOS2_USE_MPI=ON -DADIOS2_USE_Fortran=ON -DADIOS2_USE_Python=OFF -DADIOS2_BUILD_EXAMPLES_EXPERIMENTAL=OFF -DADIOS2_BUILD_TESTING=OFF -DADIOS2_USE_ADIOS1=OFF -DADIOS2_USE_HDF5=OFF -DADIOS2_USE_SST=ON ..
```

**Comment by pnorbert** [comment]
It seems you need to turn off profiling for the static build case.

-DADIOS2_USE_Profiling=OFF

**Comment by philip-davis** [comment]
I had thought I made it so that was disabled in a static build automatically, but I guess not. I will fix that.

**Comment by philip-davis** [comment]
Fixed by #1464
